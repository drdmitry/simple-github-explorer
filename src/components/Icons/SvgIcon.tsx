import type { FC } from "react";

import SvgChevronDownIcon from "./SvgChevronDownIcon";
import SvgChevronUpIcon from "./SvgChevronUpIcon";
import SvgGitHubIcon from "./SvgGitHubIcon";
import SvgSearchIcon from "./SvgSearchIcon";
import SvgStarIcon from "./SvgStarIcon";

export const ICON_NAMES = {
  ChevronDown: 'chevron-down',
  ChevronUp: 'chevron-up',
  GitHub: 'github',
  Search: 'search',
  Star: 'star',
} as const;

type ObjectValues<T> = T[keyof T];
export type IconName = ObjectValues<typeof ICON_NAMES>;

interface IProps {
  name: IconName;
  size?: number;
  className?: string;
}

export const SvgIcon: FC<IProps> = ({ name, size, className }) => {
  switch (name) {
    case ICON_NAMES.ChevronDown:
      return <SvgChevronDownIcon size={size} className={className} />;

    case ICON_NAMES.ChevronUp:
      return <SvgChevronUpIcon size={size} className={className} />;

    case ICON_NAMES.GitHub:
      return <SvgGitHubIcon size={size} className={className} />;

    case ICON_NAMES.Search:
      return <SvgSearchIcon size={size} className={className} />;

    case ICON_NAMES.Star:
      return <SvgStarIcon size={size} className={className} />;

    default:
      return null;
  };
};

export default SvgIcon;
