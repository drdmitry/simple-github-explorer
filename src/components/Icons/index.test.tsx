import React from "react";
import { render } from "@testing-library/react";

import SvgIcon from "./SvgIcon";

// import { selectionContextValue } from "./mocks/mockData";

describe("SvgIcon", () => {
  test("renders SvgIcon", () => {
    // Render the SvgIcon component
    const { container } = render(<SvgIcon name="chevron-up"/>);
    expect(container).toBeInTheDocument();

    const svgElement = container.querySelector("svg");
    expect(svgElement).toBeInTheDocument();
  });

  test("renders SvgIcon unchanged", () => {
    const { container } = render(<SvgIcon name="chevron-down" />);
    expect(container).toMatchSnapshot();
  });

  test("renders non-existing SvgIcon as null", () => {
    // @ts-ignore
    const { container } = render(<SvgIcon name="not-existing" />);
    expect(container).toBeEmptyDOMElement();
  });
});
