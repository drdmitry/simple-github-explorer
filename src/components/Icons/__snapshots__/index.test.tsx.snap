// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon renders SvgIcon unchanged 1`] = `
<div>
  <svg
    fill="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;
